### ISSUE TITLE
SVG vs BMP Costume Confusion

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap#3373]
Hi! Still working with teachers in Cleveland this week...

At the moment, we have them working through [BJC Sparks Unit 2](https://bjc.berkeley.edu/bjc-r/topic/topic.html?topic=sparks/2-sequencing-iteration.topic&course=sparks.html) Lab 2, and they are adding and manipulating costumes. Teachers figured out how to flip images horizontally, but then were quite perplexed by not being able to flip some of the images as it's not clear to newer folks what an SVG image is or why things might be different for SVGs.

Could we please have a feature to support this early-user challenge? Perhaps if someone tries to edit an SVG, there could be a popup that explains that SVGs can't be edited in Snap and offers to convert it to a BMP so they can edit it.

Thanks!

**Comment by ego-lay-atman-bay** [comment]
SVG costumes should be automatically converted to bitmap when getting pixels, right?

**Comment by maryfries** [comment]
Wow! I learn so much about how different people try to use Snap doing these trainings! I just learned (from a different teacher playing around) that you can also right-click on a sprite on the stage (not just a costume in the costumes area) to edit the costume... :)
...and that nothing happens at all if it's an SVG, which is confusing because there is no indication as to what's happening... :(

So, I do think new users need a little support here with understanding how to work with SVGs. Thanks!

**Comment by maryfries** [comment]
Hi ego-lay-atman-bay, I'm not sure what you're asking, so if that's a question for me, I'll need a little more clarification. ;)

These folks just imported built-in costumes and are finding that some are editable and some aren't, and they are confused by that.

**Comment by maryfries** [comment]
Yes, though the UI. And the other Snap veteran/curriculum developer/PD provider here with me also just got confused by this same thing, so it really is a sticking point for folks. ;)

**Comment by brianharvey** [comment]
I think this is a perfect-is-the-enemy-of-the-good issue for us.  We know we need a completely rewritten and more powerful vector costume editor, and so people think "it's not worth small improvements to something we're going to replace."  But yes, I agree, we should offer to convert to bitmap.

**Comment by jmoenig** [comment]
there is a hidden setting for that:

<img width="360" alt="rasterize" src="https://github.com/jmoenig/Snap/assets/2596285/ac59fc0f-4605-4ffd-b5eb-ed164bca999a">

But please don't tell your teachers to use it.

Listen: SVGs are - of course! - not editable as bitmaps, how could they? This is something everybody can understand, and it's very explainable. Instead they need to be edited by a special program, a vector editor. If you draw a costume using Snap's own vector editor it contains a little internal marker that tells us that you can edit it again later, because it only has elements that our editor supports. Other SVGs you take from other sites don't have that marker and should there open not in our vector editor, but instead in a little window that only lets you move their rotation center. *Some* of the "built-in" svgs have been edited by who-knows-who using who-knows-what and appear to have the same internal marker, so Snap *tries* to open the in our vector editor but isn't able to actually display them. That's a bug, and I grant y'all that it's confusing.

But please don't let them fiddle with the "rasterize" setting. Instead you can let them convert SVGs to bitmaps themselves. It's really, really easy.
